app/models/admin/delete/dogs.js

```diff
@@ -0,0 +1,110 @@
+const { breadcrumbs } = require('../../../constants/admin')
+const { formatToGds } = require('../../../lib/date-helpers')
+
+/**
+ * @param {{
+ *  column: 'status'
+ *  order: 'ASC'|'DESC'
+ * }} sort
+ * @param {string} column
+ * @return {string}
+ */
+const getAriaSort = (sort, column) => {
+  let calculatedColumn = column
+
+  if (column === undefined) {
+    calculatedColumn = 'status'
+  }
+
+  if (calculatedColumn === sort.column && sort.order === 'DESC') {
+    return 'descending'
+  }
+
+  if (calculatedColumn === sort.column) {
+    return 'ascending'
+  }
+
+  return 'none'
+}
+/**
+ * @param {{
+ *  column: 'joinedExemptionScheme'
+ *  order: 'ASC'|'DESC'
+ * }} sort
+ * @param {string} column
+ * @return {string}
+ */
+const columnLink = (sort, column) => {
+  const queryParams = new URLSearchParams()
+  let calculatedColumn = column
+
+  if (column === undefined) {
+    calculatedColumn = 'status'
+  }
+
+  const defaultSortOrder = calculatedColumn === 'interimExemptFor' ? ['DESC', 'ASC'] : ['ASC', 'DESC']
+
+  if (calculatedColumn !== 'cdoExpiry' && calculatedColumn !== 'interimExemptFor') {
+    queryParams.set('sortKey', calculatedColumn)
+  }
+
+  if (calculatedColumn === sort.column && sort.order === defaultSortOrder[0]) {
+    queryParams.set('sortOrder', defaultSortOrder[1])
+  }
+
+  const queryParamsStr = queryParams.toString()
+  const queryParamStrWithQM = queryParamsStr.length ? `?${queryParamsStr}` : ''
+
+  return queryParamStrWithQM
+}
+/**
+ * @param {Dog[]} resultList
+ * @param {{ column: string; order: 'ASC'|'DESC'}} sort
+ * @param backNav
+ * @constructor
+ */
+function ViewModel (resultList, sort, backNav) {
+  const tableHeadings = [
+    {
+      label: 'Status',
+      link: columnLink(sort, undefined),
+      ariaSort: getAriaSort(sort, undefined)
+    },
+    {
+      label: 'Index number',
+      link: columnLink(sort, 'indexNumber'),
+      ariaSort: getAriaSort(sort, 'indexNumber')
+    },
+    {
+      label: 'Date of birth',
+      link: columnLink(sort, 'dateOfBirth'),
+      ariaSort: getAriaSort(sort, 'dateOfBirth')
+    },
+    {
+      label: 'CDO issued',
+      link: columnLink(sort, 'cdoIssued'),
+      ariaSort: getAriaSort(sort, 'cdoIssued')
+    }
+  ]
+
+  const rows = [...resultList]
+  rows.forEach(row => {
+    row.humanReadableCdoIssued = formatToGds(row.cdoIssued)
+    row.humanReadableBirthDate = formatToGds(row.dateOfBirth)
+  })
```

Comment:
not important, just for info but [...variable] doesn't do a deep clone if any of the constituents of the array are objects.  It maintains link to the original object, as such line 90 - 94 is a side-affect.  It breaks 'referential transparency' and 'immutability' rules of functional programming.

Reply:
the safer way to do this would be with a mapper that does....

```
.map(row => ({
   ...row,
  humanReadableCdoIssued: formatToGds(row.cdoIssued),
  humanReadableBirthDate: formatToGds(row.dateOfBirth)
})
```